src/hooks/useAccountENSDomains.ts

```diff
@@ -21,18 +21,25 @@ async function fetchAccountENSDomains({
   accountAddress: string;
 }) {
   const result = await fetchAccountDomains(accountAddress);
-  const { domains: controlledDomains, registrations } =
-    result.data?.account || {};
+  if (!result.account) return [];
+
+  const { domains: controlledDomains, registrations } = result.account;
   const registarDomains = registrations?.map(({ domain }) => domain);
 
   const domains = uniqBy(
     [...(controlledDomains || []), ...(registarDomains || [])],
     'name'
   );
-  return domains.map(domain => {
-    prefetchENSAvatar(domain.name, { cacheFirst: true });
-    return domain;
-  });
+  return domains
+    .map(domain => {
+      if (!domain) return;
+      domain.name && prefetchENSAvatar(domain.name, { cacheFirst: true });
+      return domain;
+    })
+    .filter(
+      <TDomain>(domain: TDomain | null | undefined): domain is TDomain =>
```

Comment:
ah yes, i see the additional T on TDomain as well

Reply:
yep! generic syntax haha